app/custom_event.html

```diff
@@ -0,0 +1,136 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>RUM Custom Event Integ Test</title>
+        <script src="./loader_standard.js"></script>
+        <link
+            rel="icon"
+            type="image/png"
+            href="https://awsmedia.s3.amazonaws.com/favicon.ico"
+        />
+
+        <script>
+            // Common to all test pages
+            function dispatch() {
+                cwr('dispatch');
+            }
+
+            // Record custom event directly using API
+            function recordEvent() {
+                cwr('recordEvent', {
+                    event_type: 'custom_event_api',
+                    event_data: { customEventVersion: 255 }
+                });
+            }
+
+            // Add plugin
+            function addPlugin() {
+                class newPlugin {
+                    constructor() {
+                        this.enabled;
+                        this.context;
+                        this.id = 'custom_event_plugin';
+                    }
+                    load(context) {
+                        this.context = context;
+                    }
+                    enable() {
+                        this.enabled = true;
+                    }
+                    disable() {
+                        this.disabled = false;
+                    }
+                    getPluginId() {
+                        return this.id;
+                    }
+                    record(data) {
+                        this.context.record(this.id, data);
+                    }
+                }
+                this.plugin = new newPlugin();
+                cwr('addPlugin', this.plugin);
```

Comment:
shouldn't `addPlugin` make sure that the plugin is an instance of the plugin `Plugin` class?

Reply:
oooo we have it as an interface...

We may want to change that to make sure passed in plugins are extending the plugin class type (https://github.com/aws-observability/aws-rum-web/blob/main/src/plugins/Plugin.ts)

Reply:
It seems the best way to check if an object implements an interface is to manually do user defined type guard (checking if the object has access to all the methods), which means any Object that has the methods in the Interface will pass the type guard. Do you have suggestions for this?

Reply:
Synced up with Quinn, whether we add the type check or not, the plugin will most likely fail during `addPlugin` stage.